src/util/datastructure.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::VecDeque;
```

Comment:
Could we rename this file to something that says what it contains? Like circular_buffer.rs or such.

Reply:
Fixed